packages/jira-adapter/src/filters/fields/issues_and_projects.ts

```
@@ -0,0 +1,67 @@
+/*
+*                      Copyright 2021 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { Change, CORE_ANNOTATIONS, getChangeData, InstanceElement, isModificationChange, ObjectType } from '@salto-io/adapter-api'
+import { resolveChangeElement } from '@salto-io/adapter-utils'
+import { client as clientUtils } from '@salto-io/adapter-components'
+import { getLookUpName } from '../../references'
+
+// Works for issuesIds and projectsIds
+export const setContextField = async (
+  contextChange: Change<InstanceElement>,
+  fieldName: string,
+  endpoint: string,
+  parentField: InstanceElement,
+  client: clientUtils.HTTPWriteClientInterface
+): Promise<void> => {
+  const resolvedChange = await resolveChangeElement(contextChange, getLookUpName)
+  if (!isModificationChange(resolvedChange)) {
+    // In create the issue types and projects ids are created
+    // with the same request the context is created with
```

1. nit: just to make sure I understand, does this mean they are created as part of the request?
2. what happens on removal?
3. looks like we're not handling order changes here - do we need to ignore it to avoid potential issues? (e.g. any chance adding will do nothing and removing will remove? do we need tests for this?)

---

1. yes
2. All the values of the field are removed, no need to do anything here. Added a comment
3. If add won't work it supposed to throw an error and we won't remove (I am not sure what can be done if it will fail silently)